docs: bump actions to latest major (#600)
docker/login-action: README.md

### GitHub Container Registry

> To authenticate against the [GitHub Container Registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry),

Reviewer: ```suggestion
To authenticate to the [GitHub Container Registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry),
```

### Google Container Registry (GCR)

> [Google Artifact Registry](#google-artifact-registry-gar) is the evolution of
> Google Container Registry. As a fully-managed service with support for both
> container images and non-container artifacts. If you currently use Google
> Container Registry, use the information [on this page](https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr)
> to learn about transitioning to Google Artifact Registry. 

You can use either workload identity federation based keyless authentication or
service account based authentication.

#### Workload identity federation based authentication

> Configure the workload identity federation for github actions in gcloud (for
> steps, [refer here](https://github.com/google-github-actions/auth#setting-up-workload-identity-federation)).
> In the steps, your service account should the ability to push to GCR. Then use
> `google-github-actions/auth` action for authentication using workload identity
> like below:

Reviewer: ```suggestion
Configure the workload identity federation for GitHub Actions in Google Cloud,
[see here](https://github.com/google-github-actions/auth#setting-up-workload-identity-federation).
Your service account must have permission to push to GCR.
Use the `google-github-actions/auth` action to authenticate using workload identity
as shown in the following example:
```

  login:
    runs-on: ubuntu-latest
    steps:
    -
      name: Authenticate to Google Cloud
      id: auth
      uses: google-github-actions/auth@v1
      with:
        token_format: access_token
        workload_identity_provider: <workload_identity_provider>
        service_account: <service_account>
    -
      name: Login to GCR
      uses: docker/login-action@v3
      with:
        registry: gcr.io
        username: oauth2accesstoken
        password: ${{ steps.auth.outputs.access_token }}
```

> Replace `<workload_identity_provider>` with configured workload identity

> > provider. For steps to configure, [refer here](https://github.com/google-github-actions/auth#setting-up-workload-identity-federation).

Reviewer: ```suggestion
> provider. For steps to configure, [see here](https://github.com/google-github-actions/auth#setting-up-workload-identity-federation).
```

> Replace `<service_account>` with configured service account in workload
> identity provider which has access to push to GCR

#### Service account based authentication

> Use a service account with the ability to push to GCR and [configure access control](https://cloud.google.com/container-registry/docs/access-control).

Reviewer: ```suggestion
Use a service account with permission to push to GCR and [configure access control](https://cloud.google.com/container-registry/docs/access-control).
```

> Then create and download the JSON key for this service account and save content
> of `.json` file [as a secret](https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository)
> called `GCR_JSON_KEY` in your GitHub repo. Ensure you set the username to
> `_json_key`, or `_json_key_base64` if you use a base64-encoded key.

Reviewer: ```suggestion
Download the key for the service account as a JSON file.
Save the contents of the file [as a secret](https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository)
named `GCR_JSON_KEY` in your GitHub repository.
Set the username to `_json_key`, or `_json_key_base64` if you use a base64-encoded key.
```

### Google Artifact Registry (GAR)

> You can use either workload identity federation based keyless authentication or
> service account based authentication.

Reviewer: ```suggestion
You can authenticate with workload identity federation or a service account.
```

> #### Workload identity federation based authentication

Reviewer: ```suggestion
#### Workload identity federation
```

  login:
    runs-on: ubuntu-latest
    steps:
      -
        name: Authenticate to Google Cloud
        id: auth
        uses: google-github-actions/auth@v1
        with:
          token_format: access_token
          workload_identity_provider: <workload_identity_provider>
          service_account: <service_account>
      -
        name: Login to GAR
        uses: docker/login-action@v3
        with:
          registry: <location>-docker.pkg.dev
          username: oauth2accesstoken
          password: ${{ steps.auth.outputs.access_token }}
```

> Replace `<workload_identity_provider>` with configured workload identity
> provider

> Replace `<service_account>` with configured service account in workload
> identity provider which has access to push to GCR

> Replace `<location>` with the regional or multi-regional [location](https://cloud.google.com/artifact-registry/docs/repo-organize#locations)
> of the repository where the image is stored.

#### Service account based authentication

> Use a service account with the ability to push to GAR and [configure access control](https://cloud.google.com/artifact-registry/docs/access-control).

Reviewer: ```suggestion
Use a service account with permission to push to GAR and [configure access control](https://cloud.google.com/artifact-registry/docs/access-control).
```

> Then create and download the JSON key for this service account and save content
> of `.json` file [as a secret](https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository)
> called `GAR_JSON_KEY` in your GitHub repo. Ensure you set the username to
> `_json_key`, or `_json_key_base64` if you use a base64-encoded key.

Reviewer: ```suggestion
Download the key for the service account as a JSON file.
Save the contents of the file [as a secret](https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository)
named `GCR_JSON_KEY` in your GitHub repository.
Set the username to `_json_key`, or `_json_key_base64` if you use a base64-encoded key.
```

### AWS Public Elastic Container Registry (ECR)

> Use an IAM user with the ability to [push to ECR Public with `AmazonElasticContainerRegistryPublicPowerUser` managed policy for example](https://docs.aws.amazon.com/AmazonECR/latest/public/public-ecr-managed-policies.html#AmazonElasticContainerRegistryPublicPowerUser).

Reviewer: Seems like the link was broken

```suggestion
Use an IAM user with permission to push to ECR Public, for example using [managed policies](https://docs.aws.amazon.com/AmazonECR/latest/userguide/security-iam-awsmanpol.html#security-iam-awsmanpol-AmazonEC2ContainerRegistryPowerUser).
```

> Then create and download access keys and save `AWS_ACCESS_KEY_ID` and

Reviewer: ```suggestion
Download the access keys and save them as `AWS_ACCESS_KEY_ID` and
```

> `AWS_SECRET_ACCESS_KEY` [as secrets](https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository)

Reviewer: ```suggestion
`AWS_SECRET_ACCESS_KEY` [secrets](https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository)
```

> in your GitHub repo.

Reviewer: ```suggestion
in your GitHub repository.
```

### Quay.io

Use a [Robot account](https://docs.quay.io/glossary/robot-accounts.html) with

> the ability to push to a public/private Quay.io repository.

Reviewer: ```suggestion
permission to push to a Quay.io repository.
```

### inputs

> The following inputs can be used as `step.with` keys

Reviewer: ```suggestion
The following inputs can be used as `step.with` keys:
```

| Name       | Type   | Default | Description                                                                   |
|------------|--------|---------|-------------------------------------------------------------------------------|

> | `registry` | String |         | Server address of Docker registry. If not set then will default to Docker Hub |
> | `username` | String |         | Username used to log against the Docker registry                              |
> | `password` | String |         | Password or personal access token used to log against the Docker registry     |
> | `ecr`      | String | `auto`  | Specifies whether the given registry is ECR (`auto`, `true` or `false`)       |
> | `logout`   | Bool   | `true`  | Log out from the Docker registry at the end of a job                          |

Reviewer: ```suggestion
| `registry` | String |         | Server address of Docker registry. If not set then will default to Docker Hub |
| `username` | String |         | Username for authenticating to the Docker registry                              |
| `password` | String |         | Password or personal access for authenticating to the Docker registry     |
| `ecr`      | String | `auto`  | Specifies whether the given registry is ECR (`auto`, `true` or `false`)       |
| `logout`   | Bool   | `true`  | Log out from the Docker registry at the end of a job                          |
```
